### DISCUSSION TITLE
Squishing only certain migrations by date or range

### DISCUSSION THREAD

**Discussion by JoshSalway** [laravel/framework#43245] [Ideas]
Squishing migrations as mentioned in the Laravel docs could be improved to allow developers to squish certain migrations in a range. For example, perhaps you have old migrations you would like to squish but the newer ones you may want to keep. Also, certain migrations are useful to keep together in 1 file when you create 4 new migrations and are about to commit changes.

This command could contain flags for dates or specific migrations to merge between: 
`php artisan schema:dump --prune` 

Reference:
https://laravel.com/docs/9.x/migrations#squashing-migrations

**Comment by henzeb** [comment]
Agree. Although a simple command should be possible to allow 'everything before <date>' as that is probably the most used use case.
